### PR TITLE
adding new groups using TDD

### PR DESCRIPTION
**Pull request by ChrisCPO**

### CODE REVIEW

File: app/controllers/groups_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def show
+  end
+
+  def create
+    @group = Group.new(group_params)
+    @group.add_user(current_user)
+
+    if @group.save
```
Comment (ChrisCPO) [inline app/controllers/groups_controller.rb]: @hrs Hey so am I going crazy. In the show page here when redirected, @group show up nillclass

File: app/views/dashboards/show.html.erb [hunk 2]
```diff
@@ -1 +1,5 @@
-
+<div>
```
Comment (tabfugnic) [inline app/views/dashboards/show.html.erb:1]: Why are you surrounding this in `<div>`?

File: app/views/groups/_group.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<div class="groups">
+  <ul>
```
Comment (tabfugnic) [inline app/views/groups/_group.html.erb:2]: With this partial you will have output that would reproduce ul and lis multiple times consider moving this outside of the partial. same with the div

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -2,21 +2,14 @@
...
 
-  <% if @user.errors.any? %>
-    <%= pluralize(@user.errors.count, "error") %> prevented your account from being created:
-    <ul>
-      <% @user.errors.full_messages.each do |error_message| %>
-        <li><%= error_message %></li>
-      <% end %>
-    </ul>
-  <% end %>
+  <%= render partial: "form_errors/errors", locals: {object: @user} %>
```
Comment (tabfugnic) [inline app/views/users/new.html.erb:12]: :+1:

File: spec/features/user_groups_spec.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+feature "Group" do
+  scenario "user can create a group" do
+    user = create(:user)
+    group_name = "scotty"
+    group_description = "Beam me up"
+
+    sign_in(user)
+    visit dashboard_path
+    click_link "Create Group"
+    fill_in "Name", with: group_name
```
Comment (tabfugnic) [inline spec/features/user_groups_spec.rb:12]: This is great. Consider taking a look at https://github.com/thoughtbot/formulaic in the future

File: app/views/dashboards/show.html.erb (hunk 2 above)
Comment (ChrisCPO) [inline app/views/dashboards/show.html.erb:1]: well, my current thought process is to put everything into a div, maybe so I could center this content on the page. I was thinking about including some simple css/sass as I go, mostly just layout stuff, centering this content or something, but I have not decided one way or the other. thoughts?

**Comment by tabfugnic** [comment]
@ChrisCPO This is looking pretty good. Make a couple of changes to it and merge when ready.

Your code is really coming along. Keep it up!

**Comment by ChrisCPO** [comment]
sweet thanks @tabfugnic is formulaic a gem or part of something else that comes with suspenders, I did not see any install instructions on the githubpage.

**Pull request merged**